src/message_sender.c

```diff
@@ -690,9 +690,9 @@ static void on_link_state_changed(void* context, LINK_STATE new_link_state, LINK
         if ((message_sender->message_sender_state == MESSAGE_SENDER_STATE_OPEN) ||
             (message_sender->message_sender_state == MESSAGE_SENDER_STATE_CLOSING))
         {
-            /* User initiated transition, we should be good */
-            set_message_sender_state(message_sender, MESSAGE_SENDER_STATE_IDLE);
+            /* switch to closing so that no more requests should be accepted */
```

Comment:
I believe we once changed this because it would give more information to the upper layer when messages start to be returned back as failed.

What is the reason for changing this again?